promql/parser/printer_test.go

```diff
@@ -394,3 +398,45 @@ func TestVectorSelector_String(t *testing.T) {
 		})
 	}
 }
+
+func TestBinaryExprUTF8Labels(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{
+			name:     "UTF-8 labels in on clause",
+			input:    `foo / on("äää") bar`,
+			expected: `foo / on ("äää") bar`,
+		},
+		{
+			name:     "UTF-8 labels in group_left clause",
+			input:    `foo / on("äää") group_left("ööö") bar`,
+			expected: `foo / on ("äää") group_left ("ööö") bar`,
+		},
+		{
+			name:     "Mixed legacy and UTF-8 labels",
+			input:    `foo / on(legacy, "üüü") bar`,
+			expected: `foo / on (legacy, "üüü") bar`,
+		},
+		{
+			name:     "Legacy labels only (should not quote)",
+			input:    `foo / on(job, instance) bar`,
+			expected: `foo / on (job, instance) bar`,
+		},
```

Comment:
Thanks! This addresses the `on()` / `ignoring()`, and `group_xxx()` modifiers, but not yet the label lists for aggregators (like `sum by("äää") (bar)`, `sum without("äää") (bar)`, and so on). Could you update and add tests for those as well?